tensorflow_probability/python/distributions/categorical.py

```diff
@@ -333,6 +333,10 @@ def _entropy(self):
         _mul_exp(log_probs, log_probs),
         axis=-1)
 
+  def _mean(self):
+    probs = self.probs_parameter()
+    return tf.reduce_sum(tf.range(self._num_categories(probs),dtype=probs.dtype) * probs, axis=-1) / tf.reduce_sum(probs, axis=-1)
```

Comment:
sorry, meant [weighted_logsumexp](https://www.tensorflow.org/probability/api_docs/python/tfp/math/reduce_weighted_logsumexp)

Reply:
Should we always expect that the provided probs sum up to 1 for this method, and thus add an assertion that this is the case before computing the mean? Because other methods in the Categorical work without the sum of the probs necessarily being 1 (with validate_args=False).